lib/chef/provider/aws_ebs_volume.rb

```diff
@@ -68,4 +99,97 @@ def desired_options
       AWSResource.lookup_options(options, resource: new_resource)
     end
   end
+
+  def update_attachment(volume)
+    status = volume.status
+    #
+    # If we were told to attach the volume to a machine, do so
+    #
+    if desired_instance.is_a?(AWS::EC2::Instance)
+      case status
+      when :in_use
+        attachment = volume.attachments.first
+        # wrong instance
+        if attachment.instance != desired_instance
+          detach(volume, :instance => attachment.instance, :device => attachment.device)
+          attach(volume)
+        # wrong device only
+        elsif attachment.instance == desired_instance && attachment.device != new_resource.device
+          detach(volume, :device => attachment.device)
+          attach(volume)
+        end
+
+      when :available
+        attach(volume)
+      when nil
+        raise "EBS volume #{new_resource.name} does not currently exist!"
```

Comment:
I added it in https://github.com/chef/chef-provisioning-aws/commit/0ab222127195b536d936a551a001b7eef257450d - my output locally looks like

```
  * aws_vpc[test-vpc] (no AWS object id) action create
    - create new VPC test-vpc in us-west-2
    - create data bag item test-vpc at http://127.0.0.1:8889
    -   add reference = {"id"=>"vpc-16f34e73"}
    -   add driver_url = "aws::us-west-2"
  * aws_security_group[test-sg] (no AWS object id) action create
    - Creating new SG test-sg in us-west-2
    - create data bag item test-sg at http://127.0.0.1:8889
    -   add reference = {"id"=>"sg-5e78453b"}
    -   add driver_url = "aws::us-west-2"
  * aws_security_group[test-sg] (sg-5e78453b) action destroy
    - Deleting SG test-sg in us-west-2
    - delete data bag item test-sg at http://127.0.0.1:8889
  * aws_vpc[test-vpc] (vpc-16f34e73) action destroy
    - delete VPC test-vpc (vpc-16f34e73) in us-west-2
    - delete data bag item test-vpc at http://127.0.0.1:8889
```